torch_geometric/nn/conv/message_passing.py

```diff
@@ -48,6 +48,19 @@ class MessagePassing(torch.nn.Module):
             (default: :obj:`"source_to_target"`)
         node_dim (int, optional): The axis along which to propagate.
             (default: :obj:`-2`)
+        decomposed_layer (integer, optional): The number of feature decomposition layers.
+            (default:: int: 1). The feature decomposition method reduces the peak memory
+            usage by slicing the feature dimensions into separated feature decomposition
+            layers during GNN Aggregation. This method is disabled by default.This method
+            works well with accelerating GNN executions on CPU platforms, and can obtain
+            better results when operating on denser graphs (e.g. 2-3x speedup on Reddit dataset).
+            It can provide acceleration for common GNN models such as GCN, GIN, GraphSAGE, etc.
+            However, this method is not applicable to GNNs with attention mechanism. The selection
```

Comment:
Correct me if I am wrong, i think decomposition will work for attention based GNNs like [GATConv](https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#torch_geometric.nn.conv.GATConv) where parts of the attention are computed outside `message`. So to to be specific here we could say it's not applicable to GNNs where things like attention are computed based on `x_i` or `x_j` in the `message` function.

Reply:
I'm very sorry, my previous statement is indeed wrong, thank you for pointing out, I have revised it.